packages/@react-aria/dnd/docs/useDrag.mdx

```diff
@@ -0,0 +1,375 @@
+* `link` – indicates that there will be a relationship established between the source and target locations.
+* `cancel` – indicates that the drag and drop operation will be canceled, resulting in no changes made to the source or target.
+
+Many operating systems display these in the form of a cursor change, e.g. a plus sign to indicate a copy operation. The user may also be able to use a modifier key to choose which drop operation to perform, such as <Keyboard>Option</Keyboard> or <Keyboard>Alt</Keyboard> to switch from move to copy.
+
+The `onDragEnd` event allows the drag source to respond when a drag that it initiated ends, either because it was dropped or because it was canceled by the user. The `dropOperation` property of the event object indicates the operation that was performed. For example, when data is moved, the UI could be updated to reflect this change by removing the original dragged element.
+
+This example removes the draggable element from the UI when a move operation is completed. Try holding the <Keyboard>Option</Keyboard> or <Keyboard>Alt</Keyboard> keys to change the operation to copy, and see how the behavior changes.
+
+```tsx example
+function Draggable() {
+  let [moved, setMoved] = React.useState(false);
+  let {dragProps, isDragging} = useDrag({
+    getItems() {
+      return [{
+        'text/plain': 'hello world'
+      }];
+    },
+    /*- begin highlight -*/
+    onDragEnd(e) {
+      if (e.dropOperation === 'move') {
+        setMoved(true);
+      }
+    }
+    /*- end highlight -*/
+  });
+
+  if (moved) {
+    return null;
+  }
+
+  // ...
+  ///- begin collapse -///
+  return (
+    <div {...dragProps} role="button" tabIndex={0} className={`draggable ${isDragging ? 'dragging' : ''}`}>
+      Drag me
+    </div>
+  );
+  ///- end collapse -///
+}
+///- begin collapse -///
+<Draggable />
+<DropTarget />
+///- end collapse -///
+```
+
+The drag source can also control which drop operations are allowed for the data. For example, if moving data is not allowed, and only copying is supported, the `getAllowedDropOperations` function could be implemented to indicate this. When you drag the element below, the cursor now shows the copy affordance by default, and pressing a modifier to switch drop operations results in the drop being canceled.
+
+```tsx example
+function Draggable() {
+  let {dragProps, isDragging} = useDrag({
+    getItems() {
+      return [{
+        'text/plain': 'hello world'
+      }];
+    },
+    /*- begin highlight -*/
+    getAllowedDropOperations() {
+      return ['copy'];
+    }
+    /*- end highlight -*/
+  });
+
+  // ...
+  ///- begin collapse -///
+
+  return (
+    <div {...dragProps} className={`draggable ${isDragging ? 'dragging' : ''}`}>
+      Drag me
+    </div>
+  );
+  ///- end collapse -///
+}
+///- begin collapse -///
+<Draggable />
+<DropTarget />
+///- end collapse -///
+```
+
+## Drag affordance
```

Comment:
Is this a common term? Just wondering if 'drag button' or 'drag handle' would be easier to understand, and translate better.